src/mlpack/methods/ann/layer/perametric_relu.hpp

```diff
@@ -0,0 +1,212 @@
+/**
+ * @file perametric_relu.hpp
+ * @author Prasanna Patil
+ *
+ * Definition of PerametricReLU layer first introduced in the,
+ * Kaiming He, Xiangyu Zhang, Shaoqing, Ren Jian Sun,
+ * "Delving Deep into Rectifiers:
+ * Surpassing Human-Level Performance on ImageNet Classification", 2014
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_PERAMETRICRELU_HPP
+#define MLPACK_METHODS_ANN_LAYER_PERAMETRICRELU_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The PerametricReLU activation function, defined by (where alpha is trainable)
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& \max(x, alpha*x) \\
+ * f'(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     1 & : x > 0 \\
+ *     alpha & : x \le 0
+ *   \end{array}
+ * \right.
+ * @f}
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class PerametricReLU
+{
+ public:
+  /**
+   * Create the PerametricReLU object using the specified parameters.
+   * The non zero gradient can be adjusted by specifying tha parameter
+   * alpha in the range 0 to 1. Default (alpha = 0.03). This parameter
+   * is trainable.
+   *
+   * @param alpha Non zero gradient
+   */
+  PerametricReLU(const double user_alpha = 0.03);
```

Comment:
I would just go with alpha and use the constructor initialization list to initialize alpha.